cmd/kubectl-gadget/bcck8s.go

```diff
@@ -231,6 +231,16 @@ func bccCmd(subCommand, bccScript string) func(*cobra.Command, []string) {
 			gadgetParams = "--containersmap /sys/fs/bpf/gadget/containers"
 		}
 
+		var transform func(line string) string
+
+		if params.Columns != "" {
+			params.JsonOutput = true
```

Comment:
IIUC, even though `--columns=X,Y,Z` is only applicable to the columns output format (when `--json` is not specified), we are now asking the gadget to generate the output in JSON format because it is easier to extract the requested fields `X`, `Y` and `Z` but then, we print the output in columns, as requested. Is it right? Or did I miss something?

I noticed also that when neither `--json` nor `--columns` are specified, we are still asking the gadgets to send the output in columns format.

If my understanding is correct, what do you think about uniforming things and ask always the gadgets to send the output in JSON format and then post-process it based on the parameters:

1. When `--json` is specified: We just need to print the output as it is.
2. Otherwise: We extract the fields from a list that by default is fixed and contains all the fields of each gadget unless the `--columns` was specified.

Reply:
> IIUC, even though --columns=X,Y,Z is only applicable to the columns output format (when --json is not specified), we are now asking the gadget to generate the output in JSON format because it is easier to extract the requested fields X, Y and Z but then, we print the output in columns, as requested. Is it right? Or did I miss something?

That's right. 


> If my understanding is correct, what do you think about uniforming things and ask always the gadgets to send the output in JSON format and then post-process it based on the parameters:

That's an interesting idea but I'd prefer to delay to a different PR.